tagmanager/src/tm_ctags_wrappers.h

```diff
@@ -0,0 +1,49 @@
+/*
+ *      tm_ctags_wrappers.h - this file is part of Geany, a fast and lightweight IDE
+ *
+ *      Copyright 2016 Jiri Techet <techet(at)gmail(dot)com>
+ *
+ *      This program is free software; you can redistribute it and/or modify
+ *      it under the terms of the GNU General Public License as published by
+ *      the Free Software Foundation; either version 2 of the License, or
+ *      (at your option) any later version.
+ *
+ *      This program is distributed in the hope that it will be useful,
+ *      but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *      MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *      GNU General Public License for more details.
+ *
+ *      You should have received a copy of the GNU General Public License along
+ *      with this program; if not, write to the Free Software Foundation, Inc.,
+ *      51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+#ifndef TM_CTAGS_WRAPPERS
+#define TM_CTAGS_WRAPPERS
+
+#include <glib.h>
+
+#include "tm_parser.h"
+
+#include "entry.h" /* for sTagEntryInfo */
+
+
+G_BEGIN_DECLS
+
+typedef gboolean (*tm_ctags_callback) (const tagEntryInfo *const tag,
+	gboolean invalidate, void *user_data);
```

Comment:
OK I tried setting passCount=1 at the beginning and there are indeed a few (but really few) cases in the testing projects where the second pass is used. Actually maybe it's OK to leave the multiple passes - users of the library may just decide to ignore subsequent passes. By the way, I made the callback return boolean which is unused now but which in the future might serve as an indicator whether parsing should stop.

Maybe cleaner way than the current boolean flag or your enum would be having multiple callbacks fired at specific occasions. Two should be enough I think - the finalize one shouldn't be necessary because the caller knows when parsing finishes (it's when tm_ctags_parse() returns) and there's no need for it between passes because it would just be called immediately before the initialize step.